src/base/bittorrent/peer_filter_session_plugin.hpp

```diff
@@ -49,33 +50,45 @@ class peer_filter_session_plugin final : public lt::plugin
   {
   }
 
-  std::shared_ptr<lt::torrent_plugin> new_torrent(const lt::torrent_handle&, client_data) override
+  std::shared_ptr<lt::torrent_plugin> new_torrent(const lt::torrent_handle& th, client_data) override
   {
     // do not waste CPU and memory for useless objects when no filters are enabled
     if (!m_blacklist && !m_whitelist)
       return nullptr;
+
+    // ignore private torrents
+    if (th.torrent_file() && th.torrent_file()->priv())
+      return nullptr;
+
     return std::make_shared<peer_action_plugin>([this](auto&&... args) { return filter(args...); }, drop_peer_connection);
   }
 
 protected:
   bool filter(const lt::peer_info& info, bool handshake, bool* stop_filtering) const
   {
     if (m_blacklist) {
-      bool matched = m_blacklist->match_peer(info, false);
-      *stop_filtering = !handshake && !matched;
-      if (matched)
+      bool matched_blacklist = m_blacklist->match_peer(info, handshake);
```

Comment:
initially it was so... but it was changed to `false` (meaning "do not skip name checking" or "always check name", see corresponding function implementation for details) due to some unsupported case it was not considered initially

consider the following:
- peer with peer id `-qB4400-` (qBittorrent 4.4.0) is connected and now handshake stage
- user has the next rule in his blacklist

  ```txt
  -qB\w{4}-      Fake/MaliciousClientName
  ```

- as now is handshake stage, name (guessed by libtorrent) is ignored by filter, but given peer id matches to one of rules! result - peer is blocked on handshake by peer id without checking its name

such behavior is not desirable, blacklist should not block peers if it is "unsure", so it must check the name in any case.
if user wants to block something by peer id regardless its name, corresponding rule should be written:

```txt
-MC\d{4}-    .*
```

note the `.*` as name filter. in such case blacklist will match peer "only by peer id" regardless of its name, because `.*` matches anything

so, please revert this particular change

Reply:
Ugh, when I'm think about the logic of the second parameter passed, I was think about `false` to `not skip name`... All those boolean inversions makes me so easy to lost.

Originally I was thought about assigning the `handshake` boolean to another boolean with the name `skip_name`. But the original code caught me.

Reply:
I see this is another part that blacklist's logic should differ from the whitelist one. I'll change this back.